src/routes/authorize.rs

```diff
@@ -7,19 +7,25 @@ use tokio::sync::RwLock;
 use url::Url;
 
 use crate::{
-    constants::{OIDC_SESSION_KEY, REDIRECT_URI_SESSION_KEY, RESPONSE_TYPE_SESSION_KEY},
-    response_type::ResponseType,
-    routes::error::Error,
-    AppState,
+    constants::OIDC_SESSION_KEY, response_type::ResponseType, routes::error::Error, AppState,
 };
 
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct AuthorizeQueryParameters {
-    pub client_id: String,
+    pub client_id: Option<String>,
     pub redirect_uri: String,
-    pub response_type: String,
+    pub response_type: Option<String>,
+    pub scope: Option<String>,
+    pub state: Option<String>,
+    pub nonce: Option<String>,
+}
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct AuthorizeParameters {
```

Comment:
Can you at least add a comment to explain what this is, and how it differs from `AuthorizeQueryParameters`?

Reply:
6962ad7a128c